src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,225 @@
+                const maxValue: number = (changes.badgeMaxValue && changes.badgeMaxValue.currentValue) || this.badgeMaxValue || null;
+                this.setContent(finalText, maxValue);
+            }
+
+            // set the badge color
+            if (changes.badgeColor && changes.badgeColor.currentValue !== changes.badgeColor.previousValue) {
+                this.setBadgeColor();
+            }
+
+            // set the badge border color
+            if (changes.badgeBorderColor && changes.badgeBorderColor.currentValue !== changes.badgeBorderColor.previousValue) {
+                this.setBadgeBorderColor();
+            }
+
+            // set badge size
+            if (changes.badgeSize && changes.badgeSize.currentValue !== changes.badgeSize.previousValue) {
+                this.setBadgeSize(changes.badgeSize.previousValue);
+            }
+        }
+    }
+
+    ngOnDestroy(): void {
+        if (this._renderer.destroyNode) {
+            this._renderer.destroyNode(this._badgeElement);
+        }
+    }
+
+    private setContent(finalText: string, maxValue: number): void {
+        if (finalText && maxValue && maxValue > 0) {
+            if (this._isNumber && parseInt(finalText) > maxValue) {
+                finalText = `${maxValue}+`;
+            } else if (finalText.length > maxValue) {
+                finalText = `${finalText.substr(0, maxValue)}…`;
+            }
+        }
+
+        this._badgeDisplayContent = finalText;
+        this._badgeElement.textContent = this._badgeDisplayContent;
+    }
+
+    private setBadgeColor(): void {
+        if (this._badgeColor) {
+            this._renderer.setStyle(this._badgeElement, 'background', this._badgeColor.toHex());
+        } else {
+            this._renderer.removeStyle(this._badgeElement, 'background');
+        }
+
+        this._renderer.setStyle(this._badgeElement, 'color', this.determineContentTextColor().toHex());
+    }
+
+    private setBadgeBorderColor(): void {
+        if (this._badgeBorderColor) {
+            this._renderer.setStyle(this._badgeElement, 'border-color', this._badgeBorderColor.toHex());
+        } else {
+            this._renderer.removeStyle(this._badgeElement, 'border-color');
+        }
+    }
+
+    private setBadgeSize(previousValue: string): void {
+        if (previousValue) {
+            this._renderer.removeClass(this._badgeElement, `ux-badge-${previousValue}`);
+        }
+
+        this._renderer.addClass(this._badgeElement, `ux-badge-${this.badgeSize}`);
+    }
+
+    private determineContentTextColor(): ThemeColor {
+        return this._badgeColor
+            ? ThemeColor.parse(
+                this._contrastService.getContrastColor(this._badgeColor, this._lightColor, this._darkColor).toRgba()
+            )
+            : this._lightColor;
+    }
+
+    private parseThemeColor(color: string): ThemeColor {
+        // check for hash color
+        if (this._colorService.isHex(color)) {
```

Comment:
Can I just get some clarification about why this is needed? 

And is this a limitation as we also support rgb and rgba colors when parsing ThemeColors?

Could we do something like this just instead?

```typescript
return this._colorService.colorExists(color) ? 
    ThemeColor.parse(this._colorService.resolve(color)) : 
    ThemeColor.parse(color);
```

Reply:
The only issue with this is it doesn't allow for null.

Reply:
If I add this before your logic it will work.

```suggestion
        if (!color) {
        return null;
        }
```